### PR TITLE
To Do list App with Webpack

### PR DESCRIPTION
**Pull request by christianbiring1**
In this project, I had to build a simple HTML list of To Do tasks. The list will be styled according to the specifications listed in the lesson. This simple web page will be built using webpack and served by a webpack dev server.

## The deployed page can be seen [here]( https://christianbiring1.github.io/To-Do_list/)

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  mode: 'production',
+  entry: './src/index.js',
+  devServer: {
+    static: './dist',
+  },
+  plugins: [
+    new HtmlWebpackPlugin({
+      template: './src/index.html',
+    }),
+  ],
```
Comment (Cathella) [inline webpack.config.js:14]: ✅ HTML Webpack Plugin well setup 👍🏻.

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+const path = require('path');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+
+module.exports = {
+  mode: 'production',
```
Comment (Cathella) [inline webpack.config.js:5]: ✅ Production mode is configured 👍🏻.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "to-do_list",
+  "version": "1.0.0",
+  "description": "In this project, I had to build a simple HTML list of To Do tasks. The list will be styled according to the specifications listed in the lesson. This simple web page will be built using webpack and served by a webpack dev server.",
+  "private": true,
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "webpack",
+    "start": "webpack serve --open"
```
Comment (Cathella) [inline package.json:9]: ✅ Webpack dev server is set up and running with no errors 👍🏻.

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    description: 'Start new project',
+    completed: false,
+    index: 2,
+  },
+  {
+    description: 'Joggling',
+    completed: false,
+    index: 3,
+  },
+];
```
Comment (Cathella) [inline src/index.js:22]: ✅ It has an array of  *to do tasks* with each object containing three keys - description \[`string`\], completed \[`bool`\], index: \[`number`\]. Great job 👍🏻.

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    li.appendChild(horizontal);
+
+    const box = document.createElement('input');
+    box.setAttribute('type', 'checkbox');
+
+    const title = document.createElement('span');
+    title.classList.add('title');
+    title.innerHTML = task.description;
+    const option = document.createElement('i');
+    // option.innerHTML = '+++';
```
Comment (Cathella) [inline src/index.js]: ### 🛑 Required Change
- [x] Kindly delete commented out code. This helps in maintaining easy to read code

**Review by Cathella — CHANGES_REQUESTED** [review]
## 🟡 Status: Changes Required 🛑

### Hi @christianbiring1 👋🏻, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## To Highlight 👏🏻

✅ No linter errors.
✅ You used correct GitHub Flow.
✅ Descriptive PR.
✅ It has a function that iterates over the tasks array and populates the HTML list item element for each task.
✅ Task array is rendered dynamically to the DOM

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: webpack.config.js (hunk 2 above)
Comment (christianbiring1) [reply webpack.config.js:5]: :100: %

File: package.json (hunk 3 above)
Comment (christianbiring1) [reply package.json:9]: :100: %

File: src/index.js (hunk 4 above)
Comment (christianbiring1) [reply src/index.js:22]: :+1:

File: src/index.js (hunk 5 above)
Comment (christianbiring1) [reply src/index.js]: @Cathella  I almost forgot that I have put those stuff when I was testing my stylesheet. 
Thanks :100:

**Review by brytebee — APPROVED** [review]
Hi, @christianbiring1👋,

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**